gadget/ondisk.go

```diff
@@ -536,3 +611,51 @@ func filesystemInfo(node string) (*lsblkFilesystemInfo, error) {
 
 	return &info, nil
 }
+
+func deployFilesystemContent(ds *OnDiskStructure, gadgetRoot string) (err error) {
+	mountpoint := filepath.Join(deployMountpoint, strconv.Itoa(ds.Index))
+	if err := os.MkdirAll(mountpoint, 0755); err != nil {
+		return err
+	}
+
+	// temporarily mount the filesystem
+	if err := sysMount(ds.Node, mountpoint, ds.Filesystem, 0, ""); err != nil {
+		return fmt.Errorf("cannot mount filesystem %q at %q: %v", ds.Node, mountpoint, err)
+	}
+	defer func() {
+		errUnmount := sysUnmount(mountpoint, 0)
+		if err == nil {
+			err = errUnmount
+		}
+	}()
+
+	fs, err := NewMountedFilesystemWriter(gadgetRoot, &ds.LaidOutStructure)
+	if err != nil {
+		return fmt.Errorf("cannot create filesystem image writer: %v", err)
+	}
+
+	var preserveFiles []string
+	if err := fs.Write(mountpoint, preserveFiles); err != nil {
+		return fmt.Errorf("cannot create filesystem image: %v", err)
+	}
+
+	return nil
+}
+
+func deployNonFSContent(ds *OnDiskStructure, gadgetRoot string) error {
```

Comment:
I was wondering about this one when hiding things in the gadget. Do we need to support writing raw structures?

Reply:
This was added by @mvo5 during a refactoring, but it's not used in our current scenarios. I think it's nice to keep for completeness in case some gadget specifies nonstandard raw structures, unless someone feels strongly against it.